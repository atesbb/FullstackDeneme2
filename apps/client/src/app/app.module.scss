@import './variables.module.scss';

.app {
  font-family: sans-serif;
  margin: 1em auto;
}

body {
  background-color: $base-color-1;
  color: $text-color-1;
}

*::-webkit-scrollbar {
  width: 8px;
}

*::-webkit-scrollbar-track {
  background: $base-color-1;
}

*::-webkit-scrollbar-thumb {
  background-color: $base-color-2;
  border-radius: 20px;
  border: 2px solid $accent-color-1;
}
