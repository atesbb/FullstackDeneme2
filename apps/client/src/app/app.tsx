import styles from './app.module.scss';
import { ReactNotifications } from 'react-notifications-component';
import MainPage from './pages/MainPage';
import 'react-notifications-component/dist/theme.css';

export function App() {
  return (
    <div className={styles.app}>
      <ReactNotifications />
      <MainPage />
    </div>
  );
}

export default App;
