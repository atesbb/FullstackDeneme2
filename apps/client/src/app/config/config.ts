export const config = {
  algolia: {
    appId: process.env.NX_ALGOLIA_APP_ID as string,
    apiKey: process.env.NX_ALGOLIA_SEARCH_API_KEY as string,
    moviesIndex: process.env.NX_ALGOLIA_MOVIES_INDEX_NAME as string,
  },
};
