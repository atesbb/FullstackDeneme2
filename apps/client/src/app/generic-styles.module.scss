@import './variables.module.scss';

.inputBar {
  width: 100%;
  box-shadow: $box-shadow-1;
  border: 1.5px solid $accent-color-1;
  padding: 5px;
  height: 20px;
  border-radius: 3px;
  outline: none;
  background-color: $base-color-2;
  color: $text-color-1;
}

.button {
  background-color: $base-color-2;
  border: 2px solid $accent-color-1;
  border-radius: 3px;
  color: $text-color-1;

  &:active {
    background-color: $accent-color-1;
  }

  &:hover {
    border-color: $text-color-1;
  }
}
