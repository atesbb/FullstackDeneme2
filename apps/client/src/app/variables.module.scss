$base-color-1: #181818;
$base-color-2: #212121;

$accent-color-1: #6c757d;

$text-color-1: #fff;
$text-color-2: rgb(196, 184, 166);

$box-shadow-1: rgba(50, 50, 93, 0.25) 0px 6px 12px -2px,
  rgba(0, 0, 0, 0.3) 0px -1px 7px -3px;

$danger-color-1: rgb(172, 21, 21);
$success-color-1: rgb(22, 190, 58);
