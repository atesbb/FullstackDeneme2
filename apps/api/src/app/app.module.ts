import { Module } from '@nestjs/common';
import { MovieModule } from './modules/movie/movie.module';
import { ConfigModule } from '@nestjs/config';
import { MongooseModule } from '@nestjs/mongoose';
import { MongodbConfig } from './configs/mongodb.config';
import { ScheduleModule } from '@nestjs/schedule';
import { ConsoleModule } from 'nestjs-console';

@Module({
  imports: [
    ConfigModule.forRoot(),
    MongooseModule.forRoot(MongodbConfig.uri, {
      user: MongodbConfig.username,
      pass: MongodbConfig.password,
      dbName: MongodbConfig.dbName,
    }),
    ScheduleModule.forRoot(),
    ConsoleModule,
    MovieModule,
  ],
  controllers: [],
  providers: [],
  exports: [],
})
export class AppModule {}
