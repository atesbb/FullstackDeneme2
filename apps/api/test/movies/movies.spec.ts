import * as request from 'supertest';
import { Test } from '@nestjs/testing';
import { AppModule } from '../../src/app/app.module';
import { INestApplication, ValidationPipe } from '@nestjs/common';
import {
  MovieDocument,
  MovieModel,
  MovieSchema,
} from '../../src/app/modules/movie/movie.schema';
import { Model } from 'mongoose';
import { moviesMockData } from '@types';
import { AlgoliaService } from 'nestjs-algolia';
import { getModelToken, MongooseModule } from '@nestjs/mongoose';
import { Movie, MovieErrorCodes } from '@types';
import { AlgoliaConfig } from '../../src/app/configs/algolia.config';
import { delay } from '../utils/delay.util';
import { SearchIndex } from 'algoliasearch';
import { cleanup } from '../utils/cleanup.util';

describe('Movies', () => {
  let app: INestApplication;
  let movieModel: Model<MovieDocument>;
  let httpServer: unknown;
  let algoliaService: AlgoliaService;
  let index: SearchIndex;

  const delayTimeMs = 5000; // time to wait for indexing

  beforeAll(async () => {
    const moduleRef = await Test.createTestingModule({
      imports: [
        AppModule,
        MongooseModule.forFeature([
          { name: MovieModel.name, schema: MovieSchema },
        ]),
      ],
    }).compile();

    app = moduleRef.createNestApplication();
    app.useGlobalPipes(new ValidationPipe());
    await app.init();

    movieModel = app.get(getModelToken(MovieModel.name));
    httpServer = app.getHttpServer();
    algoliaService = app.get(AlgoliaService);
    index = algoliaService.initIndex(AlgoliaConfig.moviesIndex);
  });

  afterEach(async () => {
    await cleanup(movieModel, AlgoliaConfig.moviesIndex);
  });

  afterAll(async () => {
    await app.close();
  });

  const movieDocumentSnapshot = {
    _id: expect.any(String),
    objectID: expect.any(String),
    createdAt: expect.any(Date),
    updatedAt: expect.any(Date),
  };

  const movieIndexSnapshot = {
    _id: expect.any(String),
    objectID: expect.any(String),
    createdAt: expect.any(String),
    updatedAt: expect.any(String),
    _highlightResult: expect.any(Object),
  };

  it(`POST/movies -> should save a movie in a database and index`, async () => {
    const res = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[0])
      .expect(201);
    const uuid = res.body.uuid;

    expect(uuid).toBeDefined();

    const [movie] = await movieModel.find({ _id: uuid });

    expect(movie.toObject()).toMatchSnapshot(movieDocumentSnapshot);

    await delay(delayTimeMs); // Giving some time to Algolia to finish indexing

    const indexedMovie = await index.search(movie._id);
    expect(indexedMovie.hits.length).toBe(1);
    expect(indexedMovie.hits[0].objectID).toBe(movie.objectID);
    expect(indexedMovie.hits[0]).toMatchSnapshot(movieIndexSnapshot);
  });

  it(`POST/movies -> posting with incorrect body should give validation errors`, async () => {
    const res = await request(httpServer)
      .post(`/movies`)
      .send(undefined)
      .expect(400);
    expect(res.body).toMatchSnapshot();

    const res2 = await request(httpServer)
      .post(`/movies`)
      .send({ ...moviesMockData[0], title: 1, year: 'aa' })
      .expect(400);
    expect(res2.body).toMatchSnapshot();
  });

  it(`POST/movies -> creating a movie that already exists should give an error`, async () => {
    await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[1])
      .expect(201);

    const res = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[1])
      .expect(409);

    expect(res.body.code).toBe(MovieErrorCodes.alreadyExists);
  });

  it(`PATCH/movies -> should update a movie in a database and index`, async () => {
    const postResponse = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[2])
      .expect(201);

    const uuid = postResponse.body.uuid;
    const newTitle = 'The Godfather 2';
    const newYear = 1988;

    await request(httpServer)
      .patch(`/movies/${uuid}`)
      .send({ title: newTitle, year: newYear })
      .expect(200);

    const [movie] = await movieModel.find({ _id: uuid });

    expect(movie.title).toBe(newTitle);
    expect(movie.year).toBe(newYear);
    expect(movie.toObject()).toMatchSnapshot(movieDocumentSnapshot);

    await delay(delayTimeMs);

    const indexedMovie = await index.search(newTitle);
    expect(indexedMovie.hits.length).toBe(1);
    expect(indexedMovie.hits[0].objectID).toBe(movie.objectID);
    expect((indexedMovie.hits[0] as Movie).title).toBe(newTitle);
    expect((indexedMovie.hits[0] as Movie).year).toBe(newYear);
    expect(indexedMovie.hits[0]).toMatchSnapshot(movieIndexSnapshot);
  });

  it(`PATCH/movies -> updating a movie with an incorrect body should give validation errors`, async () => {
    const updateResponse = await request(httpServer)
      .patch(`/movies/4a5f0e2d-a272-469b-b6fb-2af552783a75`)
      .send({ title: 1, year: 'aa' })
      .expect(400);

    expect(updateResponse.body).toMatchSnapshot();
  });

  it(`PATCH/movies -> updating deleted movie should return an error`, async () => {
    const postResponse = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[2])
      .expect(201);

    const uuid = postResponse.body.uuid;
    await request(httpServer).delete(`/movies/${uuid}`).expect(200);

    const newTitle = 'The Godfather 2';

    const updateResponse = await request(httpServer)
      .patch(`/movies/${uuid}`)
      .send({ title: newTitle })
      .expect(404);

    expect(updateResponse.body.code).toBe(MovieErrorCodes.notFound);
  });

  it(`DELETE/movies -> should delete a movie in a database and index`, async () => {
    const postResponse = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[3])
      .expect(201);
    const uuid = postResponse.body.uuid;

    await request(httpServer).delete(`/movies/${uuid}`).expect(200);

    const [movie] = await movieModel.find({ _id: uuid });
    expect(movie).toBeUndefined();

    await delay(delayTimeMs);

    const indexedMovie = await index.search(uuid);
    expect(indexedMovie.hits.length).toBe(0);
  });

  it(`DELETE/movies -> deleting already deleted movie should give an error`, async () => {
    const postResponse = await request(httpServer)
      .post(`/movies`)
      .send(moviesMockData[3])
      .expect(201);
    const uuid = postResponse.body.uuid;

    await request(httpServer).delete(`/movies/${uuid}`).expect(200);
    const secondDeleteResponse = await request(httpServer)
      .delete(`/movies/${uuid}`)
      .expect(404);
    expect(secondDeleteResponse.body.code).toBe(MovieErrorCodes.notFound);
  });
});
