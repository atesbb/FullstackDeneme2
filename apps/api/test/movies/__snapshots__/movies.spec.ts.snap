// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies PATCH/movies -> should update a movie in a database and index 1`] = `
Object {
  "__v": 0,
  "_deleted": false,
  "_id": Any<String>,
  "_indexed": true,
  "actor_facets": Array [
    "https://image.tmdb.org/t/p/w45/e2u2Vyy66j2rUL8fyjjHWlYtWLH.jpg|Marlon Brando",
    "https://image.tmdb.org/t/p/w45/lk0nt14sH96SYHDkX7qsLlpqRjJ.jpg|Al Pacino",
    "https://image.tmdb.org/t/p/w45/g4bxNXWft1jLZX8gKk4G6ypkTUf.jpg|James Caan",
    "https://image.tmdb.org/t/p/w45/uBNqpxmlNSX7FJeubO1g3nlrrkF.jpg|Richard S. Castellano",
    "https://image.tmdb.org/t/p/w45/1aBC7NxPy10ofng6HsJBecJ1vMZ.jpg|Robert Duvall",
    "https://image.tmdb.org/t/p/w45/aE85Dzwaf5rTbjLnMgOtpw47BSY.jpg|Sterling Hayden",
    "https://image.tmdb.org/t/p/w45/ecteroI3S9feBlkxm33HJbPotHt.jpg|John Marley",
    "https://image.tmdb.org/t/p/w45/5682H8oxwIezxImi7HiFnBYLL4V.jpg|Richard Conte",
    "https://image.tmdb.org/t/p/w45/uTzNiDToGv513rvxQMx37mz0q9V.jpg|Al Lettieri",
    "https://image.tmdb.org/t/p/w45/fzgUMnbOkxC6E3EFcYHWHFaiKyp.jpg|Diane Keaton",
    "https://image.tmdb.org/t/p/w45/pUgFgNhow9niMPfTvj3mOzwtfRk.jpg|Abe Vigoda",
    "https://image.tmdb.org/t/p/w45/d2uI8tGSdpmXMh96C238J2A30RA.jpg|Talia Shire",
    "https://image.tmdb.org/t/p/w45/ttNwx2hFyQ7RkhAFhcfXtOYr6md.jpg|Gianni Russo",
    "https://image.tmdb.org/t/p/w45/weAjoU12KcVhIePI8YwxKkO4vbX.jpg|John Cazale",
    "https://image.tmdb.org/t/p/w45/wNpO0orqR8rWz7tf11iKN0UhvGg.jpg|Rudy Bond",
    "https://image.tmdb.org/t/p/w45/4rdC1Ezms99xSDrQnqHZDDpdZiW.jpg|Al Martino",
    "https://image.tmdb.org/t/p/w45/k9PLB28bm1oylZGAyzkWbIrZioM.jpg|Morgana King",
    "https://image.tmdb.org/t/p/w45/6rCsufBnJAj57mJ7PVaAFpIzm3j.jpg|Lenny Montana",
    "https://image.tmdb.org/t/p/w45/pq72Duoc6F6fP5rlVYE2nTU9mFH.jpg|John Martino",
    "https://image.tmdb.org/t/p/w45/grnzoe9ElGN0qgcFZDQjMN0w6oq.jpg|Salvatore Corsitto",
    "https://image.tmdb.org/t/p/w45/lISkRNmrR2P8OXvgPBZc7QmkLWA.jpg|Alex Rocco",
    "https://image.tmdb.org/t/p/w45/oqR33f9gT5ODA8U8h2y05myM7dj.jpg|Tony Giorgio",
    "https://image.tmdb.org/t/p/w45/g4syFwXi60P1ID3aLTyxup4EXbT.jpg|Victor Rendina",
    "https://image.tmdb.org/t/p/w45/oNH6iH1IkMvWJevQRtnVnXygd8b.jpg|Simonetta Stefanelli",
    "https://image.tmdb.org/t/p/w45/JjUgH6kyEyZ4kgfusXENCkhlk8.jpg|Saro Urzì",
    "https://image.tmdb.org/t/p/w45/zG1WS4sS5jvYkObHMxsK6fGK7Uk.jpg|Sofia Coppola",
    "https://image.tmdb.org/t/p/w45/ndDIYrXz2catGlP01JUwuHi2nFU.jpg|Louis Guss",
    "https://image.tmdb.org/t/p/w45/sDwZ0sbq1AHGZGuUr9IlumiVzgf.jpg|Gabriele Torrei",
  ],
  "actors": Array [
    "Marlon Brando",
    "Al Pacino",
    "James Caan",
    "Richard S. Castellano",
    "Robert Duvall",
    "Sterling Hayden",
    "John Marley",
    "Richard Conte",
    "Al Lettieri",
    "Diane Keaton",
    "Abe Vigoda",
    "Talia Shire",
    "Gianni Russo",
    "John Cazale",
    "Rudy Bond",
    "Al Martino",
    "Morgana King",
    "Lenny Montana",
    "John Martino",
    "Salvatore Corsitto",
    "Alex Rocco",
    "Tony Giorgio",
    "Victor Rendina",
    "Simonetta Stefanelli",
    "Saro Urzì",
    "Sofia Coppola",
    "Louis Guss",
    "Gabriele Torrei",
  ],
  "alternative_titles": Array [
    "The Godfather Part I",
    "El Padrino I",
    "The Godfather Part 1",
    "Mario Puzo's The Godfather",
    "The Godfather: The Coppola Restoration",
    "Ojciec Chrzestny",
    "Der Pate",
    "教父",
    "Der Pate Vol 1",
    "Крестный отец 1",
    "Der Pate Teil I",
    "The Godfather: Part I",
    "Le Parrain 1",
    "O Poderoso Chefão",
    "The Godfather 1",
    "Il Padrino - Parte I",
    "대부",
    "O Padrinho",
    "Godfather, The",
  ],
  "color": "#938071",
  "createdAt": Any<Date>,
  "genre": Array [
    "Drama",
    "Crime",
  ],
  "image": "https://image.tmdb.org/t/p/w154/d4KNaTrltq6bpkFS01pYtyXa09m.jpg",
  "objectID": Any<String>,
  "rating": 5,
  "score": 9.964740368509213,
  "title": "The Godfather 2",
  "updatedAt": Any<Date>,
  "year": 1988,
}
`;

exports[`Movies PATCH/movies -> should update a movie in a database and index 2`] = `
Object {
  "_highlightResult": Any<Object>,
  "_id": Any<String>,
  "actor_facets": Array [
    "https://image.tmdb.org/t/p/w45/e2u2Vyy66j2rUL8fyjjHWlYtWLH.jpg|Marlon Brando",
    "https://image.tmdb.org/t/p/w45/lk0nt14sH96SYHDkX7qsLlpqRjJ.jpg|Al Pacino",
    "https://image.tmdb.org/t/p/w45/g4bxNXWft1jLZX8gKk4G6ypkTUf.jpg|James Caan",
    "https://image.tmdb.org/t/p/w45/uBNqpxmlNSX7FJeubO1g3nlrrkF.jpg|Richard S. Castellano",
    "https://image.tmdb.org/t/p/w45/1aBC7NxPy10ofng6HsJBecJ1vMZ.jpg|Robert Duvall",
    "https://image.tmdb.org/t/p/w45/aE85Dzwaf5rTbjLnMgOtpw47BSY.jpg|Sterling Hayden",
    "https://image.tmdb.org/t/p/w45/ecteroI3S9feBlkxm33HJbPotHt.jpg|John Marley",
    "https://image.tmdb.org/t/p/w45/5682H8oxwIezxImi7HiFnBYLL4V.jpg|Richard Conte",
    "https://image.tmdb.org/t/p/w45/uTzNiDToGv513rvxQMx37mz0q9V.jpg|Al Lettieri",
    "https://image.tmdb.org/t/p/w45/fzgUMnbOkxC6E3EFcYHWHFaiKyp.jpg|Diane Keaton",
    "https://image.tmdb.org/t/p/w45/pUgFgNhow9niMPfTvj3mOzwtfRk.jpg|Abe Vigoda",
    "https://image.tmdb.org/t/p/w45/d2uI8tGSdpmXMh96C238J2A30RA.jpg|Talia Shire",
    "https://image.tmdb.org/t/p/w45/ttNwx2hFyQ7RkhAFhcfXtOYr6md.jpg|Gianni Russo",
    "https://image.tmdb.org/t/p/w45/weAjoU12KcVhIePI8YwxKkO4vbX.jpg|John Cazale",
    "https://image.tmdb.org/t/p/w45/wNpO0orqR8rWz7tf11iKN0UhvGg.jpg|Rudy Bond",
    "https://image.tmdb.org/t/p/w45/4rdC1Ezms99xSDrQnqHZDDpdZiW.jpg|Al Martino",
    "https://image.tmdb.org/t/p/w45/k9PLB28bm1oylZGAyzkWbIrZioM.jpg|Morgana King",
    "https://image.tmdb.org/t/p/w45/6rCsufBnJAj57mJ7PVaAFpIzm3j.jpg|Lenny Montana",
    "https://image.tmdb.org/t/p/w45/pq72Duoc6F6fP5rlVYE2nTU9mFH.jpg|John Martino",
    "https://image.tmdb.org/t/p/w45/grnzoe9ElGN0qgcFZDQjMN0w6oq.jpg|Salvatore Corsitto",
    "https://image.tmdb.org/t/p/w45/lISkRNmrR2P8OXvgPBZc7QmkLWA.jpg|Alex Rocco",
    "https://image.tmdb.org/t/p/w45/oqR33f9gT5ODA8U8h2y05myM7dj.jpg|Tony Giorgio",
    "https://image.tmdb.org/t/p/w45/g4syFwXi60P1ID3aLTyxup4EXbT.jpg|Victor Rendina",
    "https://image.tmdb.org/t/p/w45/oNH6iH1IkMvWJevQRtnVnXygd8b.jpg|Simonetta Stefanelli",
    "https://image.tmdb.org/t/p/w45/JjUgH6kyEyZ4kgfusXENCkhlk8.jpg|Saro Urzì",
    "https://image.tmdb.org/t/p/w45/zG1WS4sS5jvYkObHMxsK6fGK7Uk.jpg|Sofia Coppola",
    "https://image.tmdb.org/t/p/w45/ndDIYrXz2catGlP01JUwuHi2nFU.jpg|Louis Guss",
    "https://image.tmdb.org/t/p/w45/sDwZ0sbq1AHGZGuUr9IlumiVzgf.jpg|Gabriele Torrei",
  ],
  "actors": Array [
    "Marlon Brando",
    "Al Pacino",
    "James Caan",
    "Richard S. Castellano",
    "Robert Duvall",
    "Sterling Hayden",
    "John Marley",
    "Richard Conte",
    "Al Lettieri",
    "Diane Keaton",
    "Abe Vigoda",
    "Talia Shire",
    "Gianni Russo",
    "John Cazale",
    "Rudy Bond",
    "Al Martino",
    "Morgana King",
    "Lenny Montana",
    "John Martino",
    "Salvatore Corsitto",
    "Alex Rocco",
    "Tony Giorgio",
    "Victor Rendina",
    "Simonetta Stefanelli",
    "Saro Urzì",
    "Sofia Coppola",
    "Louis Guss",
    "Gabriele Torrei",
  ],
  "alternative_titles": Array [
    "The Godfather Part I",
    "El Padrino I",
    "The Godfather Part 1",
    "Mario Puzo's The Godfather",
    "The Godfather: The Coppola Restoration",
    "Ojciec Chrzestny",
    "Der Pate",
    "教父",
    "Der Pate Vol 1",
    "Крестный отец 1",
    "Der Pate Teil I",
    "The Godfather: Part I",
    "Le Parrain 1",
    "O Poderoso Chefão",
    "The Godfather 1",
    "Il Padrino - Parte I",
    "대부",
    "O Padrinho",
    "Godfather, The",
  ],
  "color": "#938071",
  "createdAt": Any<String>,
  "genre": Array [
    "Drama",
    "Crime",
  ],
  "image": "https://image.tmdb.org/t/p/w154/d4KNaTrltq6bpkFS01pYtyXa09m.jpg",
  "objectID": Any<String>,
  "rating": 5,
  "score": 9.964740368509213,
  "title": "The Godfather 2",
  "updatedAt": Any<String>,
  "year": 1988,
}
`;

exports[`Movies PATCH/movies -> updating a movie with an incorrect body should give validation errors 1`] = `
Object {
  "error": "Bad Request",
  "message": Array [
    "title must be a string",
    "title must be longer than or equal to 2 characters",
    "title must be shorter than or equal to 500 characters",
    "movie year cannot be in the future",
    "movie year cannot be lower than 1900",
    "year must be an integer number",
    "year must be a positive number",
  ],
  "statusCode": 400,
}
`;

exports[`Movies POST/movies -> posting with incorrect body should give validation errors 1`] = `
Object {
  "error": "Bad Request",
  "message": Array [
    "title must be a string",
    "title must be longer than or equal to 2 characters",
    "title must be shorter than or equal to 500 characters",
    "alternative_titles must be an array",
    "each value in alternative_titles must be a string",
    "each value in alternative_titles must be longer than or equal to 2 characters",
    "each value in alternative_titles must be shorter than or equal to 500 characters",
    "movie year cannot be in the future",
    "movie year cannot be lower than 1900",
    "year must be an integer number",
    "year must be a positive number",
    "image must be a string",
    "image must be an URL address",
    "image must be longer than or equal to 10 characters",
    "image must be shorter than or equal to 1000 characters",
    "color must be a hexadecimal color",
    "score must not be less than 1",
    "score must not be greater than 10",
    "score must be a positive number",
    "rating must not be less than 1",
    "rating must not be greater than 5",
    "rating must be an integer number",
    "rating must be a positive number",
    "actors must be an array",
    "each value in actors must be a string",
    "each value in actors must be longer than or equal to 2 characters",
    "each value in actors must be shorter than or equal to 100 characters",
    "actor_facets must be an array",
    "each value in actor_facets must be a string",
    "each value in actor_facets must be longer than or equal to 2 characters",
    "each value in actor_facets must be shorter than or equal to 1000 characters",
    "genre must be an array",
    "each value in genre must be a string",
    "genre must be alphanumeric",
    "each value in genre must be longer than or equal to 2 characters",
    "each value in genre must be shorter than or equal to 50 characters",
  ],
  "statusCode": 400,
}
`;

exports[`Movies POST/movies -> posting with incorrect body should give validation errors 2`] = `
Object {
  "error": "Bad Request",
  "message": Array [
    "title must be a string",
    "title must be longer than or equal to 2 characters",
    "title must be shorter than or equal to 500 characters",
    "movie year cannot be in the future",
    "movie year cannot be lower than 1900",
    "year must be an integer number",
    "year must be a positive number",
  ],
  "statusCode": 400,
}
`;

exports[`Movies POST/movies -> should save a movie in a database and index 1`] = `
Object {
  "__v": 0,
  "_deleted": false,
  "_id": Any<String>,
  "_indexed": true,
  "actor_facets": Array [
    "https://image.tmdb.org/t/p/w45/tuZCyZVVbHcpvtCgriSp5RRPwMX.jpg|Tim Robbins",
    "https://image.tmdb.org/t/p/w45/oGJQhOpT8S1M56tvSsbEBePV5O1.jpg|Morgan Freeman",
    "https://image.tmdb.org/t/p/w45/b3NfI0IzPYI40eIEtO9O0XQiR8j.jpg|Bob Gunton",
    "https://image.tmdb.org/t/p/w45/deRJUFbO8uqPSQT3B6Vgp4jiJir.jpg|William Sadler",
    "https://image.tmdb.org/t/p/w45/pwiG1ljLoqfcmFH2zFp5NP2ML4B.jpg|Clancy Brown",
    "https://image.tmdb.org/t/p/w45/f5An5NqejnTEflFmW7Vp18zVOvJ.jpg|Gil Bellows",
    "https://image.tmdb.org/t/p/w45/bsh3cqDNwVvux4NdaY1Bj4S7mNS.jpg|Mark Rolston",
    "https://image.tmdb.org/t/p/w45/r1xOgXFjqhn2fonn78rlXKPZGFw.jpg|James Whitmore",
    "https://image.tmdb.org/t/p/w45/wMRlF3VRApPduQBAuNEVM4ncYcN.jpg|Jeffrey DeMunn",
    "https://image.tmdb.org/t/p/w45/3TGsmGFwJps4dmVncOZIO3p6ToO.jpg|Larry Brandenburg",
    "https://image.tmdb.org/t/p/w45/ujBzP61tYlwqWpB3oOxknl1XuEg.jpg|David Proval",
    "https://image.tmdb.org/t/p/w45/6nuAG4DVlCc0h2rfrbpJhdmKudx.jpg|Jude Ciccolella",
  ],
  "actors": Array [
    "Tim Robbins",
    "Morgan Freeman",
    "Bob Gunton",
    "William Sadler",
    "Clancy Brown",
    "Gil Bellows",
    "Mark Rolston",
    "James Whitmore",
    "Jeffrey DeMunn",
    "Larry Brandenburg",
    "David Proval",
    "Jude Ciccolella",
  ],
  "alternative_titles": Array [
    "En verden udenfor",
    "Cadena Perpetua",
    "A remény rabjai",
    "Um Sonho de Liberdade",
    "Les évadés",
    "Rita Hayworth - Avain pakoon",
    "Rita Hayworth - nyckel till flykten",
    "À l'ombre de Shawshank",
    "Vykoupení z věznice Shawshank",
    "Vykúpenie z väznice Shawshank",
    "Nyckeln Till Frihet",
    "Esaretin bedeli",
    "Frihetens Regn",
    "Os Condenados de Shawshank",
    "肖申克的救赎",
    "Sueños de libertad",
    "Die Verurteilten",
    "Изкуплението Шоушенк",
    "Sueños de fuga",
    "Xiao shen ke de jiu shu",
    "Shawshanki lunastus",
    "Gaqceva shoushenkidan",
    "Τελευταία Έξοδος Ρίτα Χέϊγουόρθ",
    "Iskupljenje u Shawshanku",
    "Homot Shel Tikva",
    "Rastgari Dar Shawshank",
    "Shôshanku no sora ni",
    "Pabegimas is Sousenko",
    "Sueños de libertad",
    "De ontsnapping",
    "Frihetens regn",
    "Sueños de fuga",
    "Skazani na Shawshank",
    "Inchisoarea Ingerilor",
    "Побег из Шоушенка",
    "Nyckeln till frihet",
    "Kaznilnica odresitve",
    "Esaretin Bedeli",
    "Втеча з Шоушенка",
    "쇼생크 탈출",
    "The Shawshank Redemption - Stephen King",
  ],
  "color": "#8C634B",
  "createdAt": Any<Date>,
  "genre": Array [
    "Drama",
    "Crime",
  ],
  "image": "https://image.tmdb.org/t/p/w154/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
  "objectID": Any<String>,
  "rating": 5,
  "score": 9.97764206054169,
  "title": "The Shawshank Redemption",
  "updatedAt": Any<Date>,
  "year": 1994,
}
`;

exports[`Movies POST/movies -> should save a movie in a database and index 2`] = `
Object {
  "_highlightResult": Any<Object>,
  "_id": Any<String>,
  "actor_facets": Array [
    "https://image.tmdb.org/t/p/w45/tuZCyZVVbHcpvtCgriSp5RRPwMX.jpg|Tim Robbins",
    "https://image.tmdb.org/t/p/w45/oGJQhOpT8S1M56tvSsbEBePV5O1.jpg|Morgan Freeman",
    "https://image.tmdb.org/t/p/w45/b3NfI0IzPYI40eIEtO9O0XQiR8j.jpg|Bob Gunton",
    "https://image.tmdb.org/t/p/w45/deRJUFbO8uqPSQT3B6Vgp4jiJir.jpg|William Sadler",
    "https://image.tmdb.org/t/p/w45/pwiG1ljLoqfcmFH2zFp5NP2ML4B.jpg|Clancy Brown",
    "https://image.tmdb.org/t/p/w45/f5An5NqejnTEflFmW7Vp18zVOvJ.jpg|Gil Bellows",
    "https://image.tmdb.org/t/p/w45/bsh3cqDNwVvux4NdaY1Bj4S7mNS.jpg|Mark Rolston",
    "https://image.tmdb.org/t/p/w45/r1xOgXFjqhn2fonn78rlXKPZGFw.jpg|James Whitmore",
    "https://image.tmdb.org/t/p/w45/wMRlF3VRApPduQBAuNEVM4ncYcN.jpg|Jeffrey DeMunn",
    "https://image.tmdb.org/t/p/w45/3TGsmGFwJps4dmVncOZIO3p6ToO.jpg|Larry Brandenburg",
    "https://image.tmdb.org/t/p/w45/ujBzP61tYlwqWpB3oOxknl1XuEg.jpg|David Proval",
    "https://image.tmdb.org/t/p/w45/6nuAG4DVlCc0h2rfrbpJhdmKudx.jpg|Jude Ciccolella",
  ],
  "actors": Array [
    "Tim Robbins",
    "Morgan Freeman",
    "Bob Gunton",
    "William Sadler",
    "Clancy Brown",
    "Gil Bellows",
    "Mark Rolston",
    "James Whitmore",
    "Jeffrey DeMunn",
    "Larry Brandenburg",
    "David Proval",
    "Jude Ciccolella",
  ],
  "alternative_titles": Array [
    "En verden udenfor",
    "Cadena Perpetua",
    "A remény rabjai",
    "Um Sonho de Liberdade",
    "Les évadés",
    "Rita Hayworth - Avain pakoon",
    "Rita Hayworth - nyckel till flykten",
    "À l'ombre de Shawshank",
    "Vykoupení z věznice Shawshank",
    "Vykúpenie z väznice Shawshank",
    "Nyckeln Till Frihet",
    "Esaretin bedeli",
    "Frihetens Regn",
    "Os Condenados de Shawshank",
    "肖申克的救赎",
    "Sueños de libertad",
    "Die Verurteilten",
    "Изкуплението Шоушенк",
    "Sueños de fuga",
    "Xiao shen ke de jiu shu",
    "Shawshanki lunastus",
    "Gaqceva shoushenkidan",
    "Τελευταία Έξοδος Ρίτα Χέϊγουόρθ",
    "Iskupljenje u Shawshanku",
    "Homot Shel Tikva",
    "Rastgari Dar Shawshank",
    "Shôshanku no sora ni",
    "Pabegimas is Sousenko",
    "Sueños de libertad",
    "De ontsnapping",
    "Frihetens regn",
    "Sueños de fuga",
    "Skazani na Shawshank",
    "Inchisoarea Ingerilor",
    "Побег из Шоушенка",
    "Nyckeln till frihet",
    "Kaznilnica odresitve",
    "Esaretin Bedeli",
    "Втеча з Шоушенка",
    "쇼생크 탈출",
    "The Shawshank Redemption - Stephen King",
  ],
  "color": "#8C634B",
  "createdAt": Any<String>,
  "genre": Array [
    "Drama",
    "Crime",
  ],
  "image": "https://image.tmdb.org/t/p/w154/9O7gLzmreU0nGkIB6K3BsJbzvNv.jpg",
  "objectID": Any<String>,
  "rating": 5,
  "score": 9.97764206054169,
  "title": "The Shawshank Redemption",
  "updatedAt": Any<String>,
  "year": 1994,
}
`;
